src/pykeen/nn/emb.py

```diff
@@ -212,7 +261,10 @@ def get_in_canonical_shape(
         if indices is None:
             x = x.unsqueeze(dim=0)
         else:
-            x = x.unsqueeze(dim=1)
+            if indices.ndimension() == 1:
+                x = x.unsqueeze(dim=1)
+            elif indices.ndimension() > 2:
+                raise ValueError("Canonical shape is not implemented for more than 2-dimensional index tensors")
```

Comment:
why not NotImplementedError?

Reply:
It is more a problem with the canonical shape definition having only two "prefix" dimensions.

Reply:
I adopted the error message accordingly